File: C&C++/School_curriculum/DP/03Maximum_subsequence_sum/53_MaximumSubarray.cpp
#include <math.h>
#include <vector>
using namespace std;

class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        int tmp = 0, ans = INT_MIN;
        for (int i = 0; i < nums.size(); i++)
        {
            tmp = max(tmp + nums[i], nums[i]);
            ans = max(tmp, ans);
        }

        return ans;
    }
};

File: Java/Leetcode/01_Array/53_MaximumSubarray.java
package Leetcode;

class MaxSub {

    public int maxSubArray(int[] nums) {

        int len = nums.length;
        int fn = 0;
        int ans = Integer.MIN_VALUE;

        for (int i = 0; i < len; i++) {

            fn = Math.max(nums[i], fn + nums[i]);
            ans = Math.max(fn, ans);

        }

        return ans;

    }

}